mdoc/Mono.Documentation/Updater/Formatters/AttributeValueFormatter.cs

```diff
@@ -1,66 +1,66 @@
-﻿using System;
-
-using Mono.Cecil;
-
-using Mono.Documentation.Util;
+﻿using System;
 
-namespace Mono.Documentation.Updater
+using Mono.Cecil;
+
+using Mono.Documentation.Util;
+
+namespace Mono.Documentation.Updater
 {
-    /// <summary>Formats attribute values. Should return true if it is able to format the value.</summary>
-    class AttributeValueFormatter
-    {
-        public virtual bool TryFormatValue (object v, ResolvedTypeInfo type, out string returnvalue)
-        {
-            TypeReference valueType = type.Reference;
-            if (v == null)
-            {
-                returnvalue = "null";
-                return true;
-            }
-            if (valueType.FullName == "System.Type")
-            {
-                var vTypeRef = v as TypeReference;
-                if (vTypeRef != null)
-                    returnvalue = "typeof(" + NativeTypeManager.GetTranslatedName (vTypeRef) + ")"; // TODO: drop NS handling
-                else
-                    returnvalue = "typeof(" + v.ToString () + ")";
-
-                return true;
-            }
-            if (valueType.FullName == "System.String")
-            {
-                returnvalue = "\"" + MDocUpdater.FilterSpecialChars (v.ToString ()) + "\"";
-                return true;
-            }
-            if (valueType.FullName == "System.Char")
-            {
-                returnvalue = "'" + MDocUpdater.FilterSpecialChars (v.ToString ()) + "'";
-                return true;
-            }
-            if (v is Boolean)
-            {
-                returnvalue = (bool)v ? "true" : "false";
-                return true;
-            }
-
-            TypeDefinition valueDef = type.Definition;
-            if (valueDef == null || !valueDef.IsEnum)
-            {
-                returnvalue = v.ToString ();
-                return true;
-            }
-
-            string typename = MDocUpdater.GetDocTypeFullName (valueType);
-            var values = MDocUpdater.GetEnumerationValues (valueDef);
-            long c = MDocUpdater.ToInt64 (v);
-            if (values.ContainsKey (c))
-            {
-                returnvalue = typename + "." + values[c];
-                return true;
-            }
-
-            returnvalue = null;
-            return false;
-        }
-    }
+    /// <summary>Formats attribute values. Should return true if it is able to format the value.</summary>
+    class AttributeValueFormatter
+    {
+        public virtual bool TryFormatValue (object v, ResolvedTypeInfo type, out string returnvalue)
+        {
+            TypeReference valueType = type.Reference;
+            if (v == null)
+            {
+                returnvalue = "default";

```

Comment:
ok ... so on first glance, this looks fantastic, great job 👏 Unfortunately on review I'm realizing that this fixes the issue in one language while introducing a regression for the other languages 😊

`default` is only a C# language construct ... it doesn't exist in the other languages. So instead of making the change here in this class, you should make the change in the `CSharpFullMemberFormatter` class. Look in the `AppendParameter` method there, and you will find a line that calls `MDocUpdater.MakeAttributesValueString`. You can probably add the logic in that section to call that method, and if it's null, switch it to `default`.